Figshare.js

```diff
@@ -109,14 +109,19 @@ function scrape(doc, url) {
 			//place DOI in DOI field for journal articles, into Extra otherwise
 			if (DOI) {
 				DOI = DOI.replace(/^doi:?\s*/i, "")
-				if (item.itemType == "journalArticle") {
+				if (ZU.fieldIsValidForType("DOI", item.itemType)) {
 					item.DOI = DOI;
 				}
-				else if (DOI) {
-					item.extra = "DOI:" + DOI;
+				else {
+					if (item.extra) {
+						if (item.extra.search(/^DOI:/) == -1) {
+							item.extra += "\nDOI: " + DOI;
+						}
+					} else {
+						item.extra = "DOI: " + DOI;
+					}
```

Comment:
turns out the whole passage isn't necessary -- the DOI is in the RIS and the RIS translator does handle this now.